caffe2/proto/torch.proto

```diff
@@ -4,547 +4,85 @@ import "caffe2/proto/caffe2.proto";
+
+  // static graph
+  optional caffe2.NetDef graph = 2;
+  // method is represented as torch script
+  optional string torch_script = 3;
+
+  // the names of inputs and outputs
+  repeated string inputs = 4;
+  repeated string outputs = 5;
+
+  // method type, can be one of the following:
+  //   1) MAIN, the main method, by default is `forward` method
+  //   2) INIT, only run once for initialization
+  //   3) REGULAR, may run multiple times at runtime
+  enum MethodType {
+    MAIN = 1;
+    INIT = 2;
+    REGULAR = 3;
   }
 
-  // The name field MUST be present for this version of the IR.
-  optional string name = 1;           // namespace Attribute
-
-  // if ref_attr_name is not empty, ref_attr_name is the attribute name in parent function.
-  // In this case, this AttributeProto does not contain data, and it's a reference of attribute
-  // in parent scope.
-  // NOTE: This should ONLY be used in function (sub-graph). It's invalid to be used in main graph.
-  optional string ref_attr_name = 21;
-
-  // A human-readable documentation for this attribute. Markdown is allowed.
-  optional string doc_string = 13;
+  optional MethodType method_type = 6;
 
-  // The type field MUST be present for this version of the IR.
-  // For 0.0.1 versions of the IR, this field was not defined, and
-  // implementations needed to use has_field hueristics to determine
-  // which value field was in use.  For IR_VERSION 0.0.2 or later, this
-  // field MUST be set and match the f|i|s|t|... field in use.  This
-  // change was made to accomodate proto3 implementations.
-  optional AttributeType type = 20;   // discriminator that indicates which field below is in use
+  optional string debug_info = 7;
 
-  // Exactly ONE of the following fields must be present for this version of the IR
-  optional float f = 2;               // float
-  optional int64 i = 3;               // int
-  optional bytes s = 4;               // UTF-8 string
-  optional TensorProto t = 5;         // tensor value
-  optional GraphProto g = 6;          // graph
-  // Do not use field below, it's deprecated.
-  // optional ValueProto v = 12;         // value - subsumes everything but graph
-
-  repeated float floats = 7;          // list of floats
-  repeated int64 ints = 8;            // list of ints
-  repeated bytes strings = 9;         // list of UTF-8 strings
-  repeated TensorProto tensors = 10;  // list of tensors
-  repeated GraphProto graphs = 11;    // list of graph
-}
-
-// Defines information on value, including the name, the type, and
-// the shape of the value.
-message ValueInfoProto {
-  // This field MUST be present in this version of the IR.
-  optional string name = 1;     // namespace Value
-  // This field MUST be present in this version of the IR.
-  optional TypeProto type = 2;
-  // A human-readable documentation for this value. Markdown is allowed.
-  optional string doc_string = 3;
+  repeated caffe2.Argument annotations = 8;
```

Comment:
Redundant in the case of torch script.

Reply:
Will remove this field. Originally reserved for torch script mode. NetDef already has `annotations`.